The default libdts builds without PIC support, so VLC has to use it builtin,
and looks instead for a libdts_pic to build the shared version. In Gentoo,
libdts is always built with PIC support instead, as it uses libtool, so this
patch makes sure that VLC uses the plugin version of libdts instead.
Index: vlc-0.8.5-test2/configure.ac
===================================================================
--- vlc-0.8.5-test2.orig/configure.ac
+++ vlc-0.8.5-test2/configure.ac
@@ -2914,21 +2914,16 @@ if test "${enable_dts}" != "no"; then
       VLC_ADD_CPPFLAGS([dtstofloat32],[-I${real_dts_tree}/include])
       VLC_ADD_LDFLAGS([dtstofloat32],[-L${real_dts_tree}/libdts])
       LDFLAGS="${LDFLAGS_save} ${LDFLAGS_dtstofloat32}"
-      AC_CHECK_LIB(dts_pic, dts_free, [
+      AC_CHECK_LIB(dts, dts_free, [
         VLC_ADD_PLUGINS([dtstofloat32])
-        VLC_ADD_LDFLAGS([dtstofloat32],[-ldts_pic])
+        VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
         ],[
-        AC_CHECK_LIB(dts, dts_free, [
-          VLC_ADD_BUILTINS([dtstofloat32])
-          VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
-          ],[
-          if test -f ${real_dts_tree}/libdts/libdts.a
-          then
-            AC_MSG_ERROR([make sure you have at least libdts-0.0.2])
-          else
-            AC_MSG_ERROR([the specified tree hasn't been compiled])
-          fi
-        ])
+        if test -f ${real_dts_tree}/libdts/libdts.a
+        then
+          AC_MSG_ERROR([make sure you have at least libdts-0.0.2])
+        else
+          AC_MSG_ERROR([the specified tree hasn't been compiled])
+        fi
       ])
       LDFLAGS="${LDFLAGS_save}"
     else
@@ -2938,18 +2933,13 @@ if test "${enable_dts}" != "no"; then
   else
     LDFLAGS="${LDFLAGS_save} ${LDFLAGS_dtstofloat32}"
     AC_CHECK_HEADERS(dts.h, [
-      AC_CHECK_LIB(dts_pic, dts_free, [
+      AC_CHECK_LIB(dts, dts_free, [
         VLC_ADD_PLUGINS([dtstofloat32])
-        VLC_ADD_LDFLAGS([dtstofloat32],[-ldts_pic])
+        VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
       ],[
-        AC_CHECK_LIB(dts, dts_free, [
-          VLC_ADD_BUILTINS([dtstofloat32])
-          VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
-        ],[
-          if test "${enable_dts}" = "yes"; then
-            AC_MSG_ERROR([Could not find libdts on your system: you may get it from http://www.videolan.org/dtsdec.html])
-          fi
-        ])
+        if test "${enable_dts}" = "yes"; then
+          AC_MSG_ERROR([Could not find libdts on your system: you may get it from http://www.videolan.org/dtsdec.html])
+        fi
       ])
     ])
     LDFLAGS="${LDFLAGS_save}"
